contexts/pathogen-context/pathogen-contexts/sarscov2/modelled-sarscov2-seroprevalence-context.tsx

```diff
@@ -0,0 +1,170 @@
+"use client";
+import uniq from 'lodash/uniq';
+import { createContext, useContext, useEffect, useMemo, useState } from "react";
+import { SarsCov2Context, SarsCov2Estimate } from "./sc2-context";
+import { useBestFitCurve } from "@/components/customs/visualizations/line-fitting/use-best-fit-curve";
+import { generateRange, typedGroupBy, typedObjectEntries, typedObjectFromEntries, typedObjectKeys } from "@/lib/utils";
+import { dateToMonthCount } from "@/lib/time-utils";
+import parseISO from "date-fns/parseISO";
```

Comment:
Large portions of this file are copy-pasted from `modelled-seroprevalence-by-who-region.tsx`. There's definitely a way to consolidate our code a little bit, it's just not coming to me right now. It's a bit annoying because of how the best fit line chart is set up. I want to do a larger refactor of that component because it's a bit of a pain to work with but this will have to do for now.

Reply:
This is a context because I want this data to be shared by both components, I just don't have any great ideas for how to get there as of right now.